File: test_register.js
const axios = require('axios');

// 测试注册功能的错误处理
async function testRegister() {
  console.log('开始测试注册功能的错误处理...');
  
  try {
    // 测试1: 使用已存在的用户名
    console.log('\n测试1: 使用已存在的用户名');
    const response1 = await axios.post('http://localhost:8001/auth/register', {
      username: 'testuser',
      email: 'newemail@example.com',
      phone: '13100001111',
      password: 'Test@1234',
      display_name: 'Test User'
    });
    console.log('响应:', response1.data);
  } catch (error) {
    console.log('预期的错误:', error.response?.data);
    console.log('错误消息:', error.message);
  }
  
  try {
    // 测试2: 使用已存在的邮箱
    console.log('\n测试2: 使用已存在的邮箱');
    const response2 = await axios.post('http://localhost:8001/auth/register', {
      username: 'newuser123',
      email: 'test@example.com',
      phone: '13100002222',
      password: 'Test@1234',
      display_name: 'New User'
    });
    console.log('响应:', response2.data);
  } catch (error) {
    console.log('预期的错误:', error.response?.data);
    console.log('错误消息:', error.message);
  }
  
  try {
    // 测试3: 使用弱密码
    console.log('\n测试3: 使用弱密码');
    const response3 = await axios.post('http://localhost:8001/auth/register', {
      username: 'weakpassuser',
      email: 'weakpass@example.com',
      phone: '13100003333',
      password: 'weakpass',
      display_name: 'Weak Pass User'
    });
    console.log('响应:', response3.data);
  } catch (error) {
    console.log('预期的错误:', error.response?.data);
    console.log('错误消息:', error.message);
  }
}

testRegister().catch(console.error);

File: test_register.py
import requests
import json
import uuid

# 生成唯一的测试用户数据以避免冲突
test_suffix = str(uuid.uuid4())[:8]
# 生成随机手机号（模拟）
random_phone = f"13{test_suffix}"
new_user = {
    "username": f"testuser_{test_suffix}",
    "email": f"test_{test_suffix}@example.com",
    "phone": random_phone,  # 使用随机手机号避免冲突
    "password": "password123"
}

# 测试登录功能
def test_login(identifier, password):
    url = "http://localhost:8002/auth/login"
    headers = {"Content-Type": "application/json"}
    login_data = {"identifier": identifier, "password": password}
    
    print(f"\n=== Testing Login with {identifier} ===")
    print(f"URL: {url}")
    print(f"Login Data: {json.dumps(login_data, ensure_ascii=False)}")
    
    try:
        response = requests.post(url, headers=headers, json=login_data)
        print(f"Response Status Code: {response.status_code}")
        
        try:
            response_data = response.json()
            print(f"Response Body: {json.dumps(response_data, ensure_ascii=False, indent=2)}")
        except json.JSONDecodeError:
            print(f"Response Body (not JSON): {response.text}")
            
        if response.status_code == 200:
            print("✅ Login Successful!")
        else:
            print("❌ Login Failed")
            
    except requests.exceptions.RequestException as e:
        print(f"Login Request Failed: {e}")

# 测试注册功能
def test_registration():
    url = "http://localhost:8002/auth/register"
    headers = {"Content-Type": "application/json"}
    
    print("\n=== Testing User Registration ===")
    print(f"URL: {url}")
    print(f"Headers: {headers}")
    print(f"User Data: {json.dumps(new_user, ensure_ascii=False, indent=2)}")
    
    try:
        response = requests.post(url, headers=headers, json=new_user)
        print(f"\nResponse Status Code: {response.status_code}")
        print(f"Response Headers: {dict(response.headers)}")
        
        try:
            response_data = response.json()
            print(f"Response Body: {json.dumps(response_data, ensure_ascii=False, indent=2)}")
        except json.JSONDecodeError:
            print(f"Response Body (not JSON): {response.text}")
            
        return response
    except requests.exceptions.RequestException as e:
        print(f"\nRequest Failed: {e}")
        return None

# 运行测试
response = test_registration()

# 分析结果
if response:
    if response.status_code == 201:
        print("\n✅ Registration Successful!")
        # 如果注册成功，立即测试登录
        test_login(response.json()['user']['username'], new_user['password'])
        test_login(new_user['email'], new_user['password'])
        test_login(new_user['phone'], new_user['password'])
    elif response.status_code == 400:
        print("\n❌ Registration Failed (Bad Request)")
        try:
            error_data = response.json()
            print(f"Error Message: {error_data.get('detail', 'Unknown error')}")
        except:
            print(f"Error: {response.text}")
    else:
        print(f"\n❌ Registration Failed with Status Code: {response.status_code}")
